test/sharness/t0252-files-gc.sh

```diff
@@ -0,0 +1,36 @@
+#!/bin/sh
+#
+# Copyright (c) 2016 Jeromy Johnson
+# MIT Licensed; see the LICENSE file in this repository.
+#
+
+test_description="test how the unix files api interacts with the gc"
+
+. lib/test-lib.sh
+
+test_init_ipfs
+
+test_expect_success "object not removed after gc" '
+  echo "hello world"  | ipfs files write --create /hello.txt &&
+  ipfs repo gc &&
+  ipfs cat QmVib14uvPnCP73XaCDpwugRuwfTsVbGyWbatHAmLSdZUS
```

Comment:
@kevina want to take this one on and add a few more tests?

- Test that the read test below outputs the right text
- Exercise a situation that uses the best effort pin: 
  - Add a directory containing a file with `--pin=false`
  - pin the hash directly
  - run a gc so that file is no longer accessible
  - `ipfs files cp <dirhash> /somewhere` (this should work just fine)
  - `ipfs repo gc` should succeed
  - re-add the file that was originally in that directory with `--pin=false`
  - run `ipfs repo gc` again, that file shouldnt be removed.